File: Pemograman Berorientasi Objek/Praktikum 07/Soal 6/Memo.java
/* NIM/Nama	 : 13517137/Vincent Budianto
 * Nama File : Memo.java
 * Topik     : Praktikum 07
 * Tanggal   : 14 Maret 2019
 * Deskripsi : Class Printer */

// sebuah kelas menyimpan sebuah value
// lengkapilah dengan package yang diperlukan

class Memo
{
	private int val;

	// ctor
	public Memo(int _val)
	{
		val = _val;
	};

	// getter
	public int getVal()
	// mengirim nilai val
	{
		return val;
	};

	// setter
	public void setVal(int _val)
	// set nilai val
	{
		val = _val;
	};
	
	// fungsi lain, jika anda tak sempat membuat dg benar, tutup sebagai komentar
	public void mirror()
	// "membalik" atribut val, misalnya 123 menjadi 321
	{
		int a = 0;
		
		while (val != 0)
		{
			a = a * 10;
			a = a + val%10;
			val = val/10;
		}
		
		val = a;
	};

	public String toString()
	// membentuk String dari angka 
	// misalnya val=8 dituliskan 8
	{
		return (Integer.toString(val));
	};
}

File: Pemograman Berorientasi Objek/Praktikum 06/Soal 3/memo.cpp
/* NIM/Nama  : 13517137/Vincent Budianto
 * Nama file : memo.cpp
 * Topik     : Praktikum 6
 * Tanggal   : 28 Februari 2019
 * Deskripsi : Realisasi memo.cpp */

#include "exception.h"
#include "memo.h"
#include <iostream>
using namespace std;

Memo::Memo()
{
	pesan = "";
	untuk = "";
}

Memo::Memo(string _pesan, string _untuk)
{
	if (pesan.length() <= PESAN_MAX)
	{
		pesan = _pesan;
		untuk = _untuk;
	}
	else
	{
		throw PesanKepanjanganException();
	}
}

Memo& Memo::operator=(Memo& obj)
{
	pesan = obj.pesan;
	untuk = obj.untuk;

	return *this;
}

string Memo::getPesan()
{
	return pesan;
}

string Memo::getUntuk()
{
	return untuk;
}
